libmatrix.c

```diff
@@ -139,17 +140,24 @@ static char *matrixprpl_get_chat_name(GHashTable *components)
  * Handle a double-click on a chat in the buddy list, or acceptance of a chat
  * invite: it is expected that we join the chat.
  */
-static void matrixprpl_join_chat(PurpleConnection *gc, GHashTable *components)
+static void matrixprpl_join_chat(PurpleConnection *pc, GHashTable *components)
 {
-    const char *room = g_hash_table_lookup(components, PRPL_CHAT_INFO_ROOM_ID);
-    int chat_id = g_str_hash(room);
-    PurpleConversation *conv;
+    const char *room_alias = components ?
+            g_hash_table_lookup(components, PRPL_CHAT_INFO_ROOM_ID) : "";
+    int chat_id = room_alias ? g_str_hash(room_alias) : 0;
+    PurpleConversation *conv = NULL;
     PurpleConvChat *chat;
+    MatrixConnectionData *conn = purple_connection_get_protocol_data(pc);
 
-    conv = purple_find_chat(gc, chat_id);
+    if (!chat_id) {
+        return;
+    }
 
-    if(!conv) {
-        matrix_connection_join_room(gc, room, components);
+    conv = purple_find_chat(pc, chat_id);
+
+    if (!conv) {
+        /* get the room by alias, if possible - join it, otherwise create it */
```

Comment:
why is this the right thing to do? Doesn't pidgin have different commands for "join a room" vs "create a room"?

In any case, surely we shouldn't try to create the room on *any* error?